internal/transport/http2_server.go

```diff
@@ -361,15 +362,12 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			rstCode:  http2.ErrCodeFrameSize,
 			onWrite:  func() {},
 		})
-		return false
+		return nil
 	}
 
 	if streamID%2 != 1 || streamID <= t.maxStreamID {
 		// illegal gRPC stream id.
-		if logger.V(logLevel) {
-			logger.Errorf("transport: http2Server.HandleStreams received an illegal stream id: %v", streamID)
-		}
-		return true
+		return fmt.Errorf("transport: http2Server.HandleStreams received an illegal stream id: %v", streamID)
```

Comment:
Who uses this error? Is `http2Server.HandleStreams` in the error string required/useful?

Reply:
Thanks.

Given that the log line in `Close()` looks like this:
```golang
	logger.Infof("Closing transport, will close the connection. Err: %v", err)
```
we would have a log message which looks like:
`Closing transport, will close the connection. Err: transport: http2Server.HandleStreams received an illegal stream id: <>"`

This still seems very verbose.
`Closing transport: received an illegal stream id: <>"`
conveys everything that needs to be said.

Reply:
Weird I forgot to change this in last commit, will get it out in next one.